### PR TITLE
Added unit test

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Added unit test

**Related issue(s)**

- https://github.com/kyma-project/eventing-manager/issues/8

### CODE REVIEW

File: internal/controller/eventing/integrationtests/controller/integration_test.go [hunk 1]
```diff
@@ -481,6 +481,15 @@ func Test_WatcherEventingCRK8sObjects(t *testing.T) {
 
 			// create Eventing CR
 			testEnvironment.EnsureK8sResourceCreated(t, tc.givenEventing)
+
+			defer func() {
```
Comment (mfaizanse) [inline internal/controller/eventing/integrationtests/controller/integration_test.go:485]: added to cleanup the resources created by tests.

File: internal/controller/eventing/nats_test.go [hunk 2]
```diff
@@ -96,6 +98,31 @@ func Test_reconcileNATSSubManager(t *testing.T) {
...
 			wantAssertCheck: true,
 		},
+		{
+			name: "it should retry to start subscription manager when it failed to start natsSubManager but the " +
+				"initialization of natsSubManager was successful",
+			givenIsNATSSubManagerStarted: false,
+			givenNATSSubManagerMock: func() *ecsubmanagermocks.Manager {
+				jetStreamSubManagerMock := new(ecsubmanagermocks.Manager)
+				jetStreamSubManagerMock.On("Init", mock.Anything).Return(nil).Once()
+				jetStreamSubManagerMock.On("Start", mock.Anything, mock.Anything).Return(errors.New("failed to start")).Twice()
```
Comment (mfaizanse) [inline internal/controller/eventing/nats_test.go:108]: The `.Twice()` will verify that the start was called again on retry.

**Review by grischperl — CHANGES_REQUESTED** [review]

File: internal/controller/eventing/nats_test.go [hunk 3]
```diff
@@ -96,6 +98,31 @@ func Test_reconcileNATSSubManager(t *testing.T) {
 			},
 			wantAssertCheck: true,
 		},
+		{
+			name: "it should retry to start subscription manager when it failed to start natsSubManager but the " +
+				"initialization of natsSubManager was successful",
```
[the remark this answers is not in the document]
Comment (mfaizanse) [reply internal/controller/eventing/nats_test.go]: updated!

**Review by grischperl — APPROVED** [review]

**Pull request merged**